Kernel cache drop

bengland2 opened the pull request:
This cherrypy-based web service allows ripsaw benchmarks to request a kernel cache drop from a privileged pod without themselves being privileged.     The DaemonSet yaml will be part of ripsaw so that any storage benchmark can start and invoke this service.   Tested with minikube v1.11.0 = k8s v1.18.

bengland2 committed:
basics of a web-service kernel cache dropper

still needs to be a daemonset

bengland2 committed:
more descriptive name

bengland2 committed:
DaemonSet for kernel cache dropping

bengland2 committed:
restrict pod visibility to /proc/sys/vm

bengland2 committed:
need cherrypy to run

bengland2 committed:
This file is moving to ripsaw so that benchmarks can invoke this

jtaleric commented on kernel_cache_dropper/kernel-cache-drop-websvc.py:
```diff
@@ -0,0 +1,45 @@
+    config = { 
+        'global': {
+            'server.socket_host': '0.0.0.0' ,
+            'server.socket_port': 9435,
```
Want to take this in as a env var that way the user can overwrite it? Default to 9435, but allow user to provide a means to overwrite?

bengland2 replied:
I agree, had started that but haven't finished it yet.

jtaleric reviewed (approved):
LGTM, just one suggestion, not worth blocking.

rsevilla87 commented on kernel_cache_dropper/kernel-cache-drop-websvc.py:
```diff
@@ -0,0 +1,45 @@
+                dcf.write('3\n')
+            logger.info('completed cache drop')
+        except FileNotFoundError:
+                logger.error('/proc_sys_vm/drop_caches not found')
```
This one is returning 200 OK even if /proc_sys_vm/drop_caches doesn't exist.

bengland2 replied:
@rsevilla87  wow, it never occurred to me that it wouldn't exist.   I was only concerned that the pod wouldn't have access to it.   But if the file doesn't exist, then the exception won't be caught and the program will exit.    Are you sure that you passed the pod IP to it?   request_cache_drop.py does absolutely nothing unless it has pod IPs passed to it.

rsevilla87 replied:
It always "return 'SUCCESS'" regardless a exception is caught or not :)

jtaleric replied:
I think the suggestion is just to pass a 404 if we hit the `FileNotFoundError`

rsevilla87 reviewed (changes requested):
I think we should verify that the script is running as root, or handle that exception.

bengland2 committed:
make service port number configurable with env.var.
also remove exception handler for file-not-found
add correct check for /proc filesystem

rsevilla87 commented:
LGTM!
